internal/hostnetwork/ovs_bridge.go

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package hostnetwork
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	libovsclient "github.com/ovn-kubernetes/libovsdb/client"
+	"github.com/ovn-kubernetes/libovsdb/model"
+	"github.com/ovn-kubernetes/libovsdb/ovsdb"
+	"github.com/vishvananda/netlink"
+)
+
+// Bridge model represents a row in the Bridge table
+type Bridge struct {
+	UUID        string            `ovsdb:"_uuid"`
+	Name        string            `ovsdb:"name"`
+	Ports       []string          `ovsdb:"ports"`
+	ExternalIds map[string]string `ovsdb:"external_ids"`
+}
+
+// Port model represents a row in the Port table
+type Port struct {
+	UUID       string   `ovsdb:"_uuid"`
+	Name       string   `ovsdb:"name"`
+	Interfaces []string `ovsdb:"interfaces"`
+}
+
+// Interface model represents a row in the Interface table
+type Interface struct {
+	UUID string `ovsdb:"_uuid"`
+	Name string `ovsdb:"name"`
+	Type string `ovsdb:"type"`
+}
+
+// OpenVSwitch model represents a row in the Open_vSwitch table
+type OpenVSwitch struct {
+	UUID    string   `ovsdb:"_uuid"`
+	Bridges []string `ovsdb:"bridges"`
+}
+
+func newOVSClient(ctx context.Context) (libovsclient.Client, error) {
+	dbModel, err := model.NewClientDBModel("Open_vSwitch", map[string]model.Model{
+		"Open_vSwitch": &OpenVSwitch{},
+		"Bridge":       &Bridge{},
+		"Port":         &Port{},
+		"Interface":    &Interface{},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to build OVS DB model: %w", err)
+	}
+
+	ovs, err := libovsclient.NewOVSDBClient(
+		dbModel,
+		libovsclient.WithEndpoint("unix:/var/run/openvswitch/db.sock"), // TODO: must parametrize this from the config
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OVSDB client: %w", err)
+	}
+	if err := ovs.Connect(ctx); err != nil {
+		return nil, fmt.Errorf("failed to connect to OVSDB: %w", err)
+	}
+	return ovs, nil
+}
+
+func ensureOVSBridgeAndAttach(ctx context.Context, bridgeName, ifaceName string) error {
+	slog.Info("ensureOVSBridgeAndAttach", "bridge", bridgeName, "interface", ifaceName)
+
+	// Verify the interface exists before trying to attach to OVS
+	link, err := netlink.LinkByName(ifaceName)
+	if err != nil {
+		return fmt.Errorf("interface %s does not exist on host (required before attaching to OVS bridge): %w", ifaceName, err)
+	}
+	slog.Info("interface exists on host", "name", ifaceName, "index", link.Attrs().Index, "type", link.Type())
+
+	ovs, err := newOVSClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer ovs.Close()
+
+	return ensureOVSBridgeAndAttachWithClient(ctx, ovs, bridgeName, ifaceName)
```

Comment:
how about merging this function and the on below?

Reply:
What's the benefit ?

I think this split provides clean separation of responsibilities, and makes it easier to unit test.